test/Conversion/LLHDToLLVM/convert_signals.mlir

```diff
@@ -115,27 +115,37 @@ llhd.entity @convert_drv (%sI1 : !llhd.sig<i1>, %sArr : !llhd.sig<!llhd.array<3x
 // CHECK-SAME:                                  %[[VAL_2:.*]]: !llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>) {
 // CHECK:           %[[VAL_3:.*]] = llvm.mlir.constant(0 : i32) : !llvm.i32
 // CHECK:           %[[VAL_4:.*]] = llvm.getelementptr %[[VAL_2]]{{\[}}%[[VAL_3]]] : (!llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>, !llvm.i32) -> !llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>
-// CHECK:           %[[VAL_5:.*]] = llvm.mlir.constant(false) : !llvm.i1
-// CHECK:           %[[VAL_6:.*]] = llvm.mlir.constant(dense<[1000, 0, 0]> : vector<3xi64>) : !llvm.array<3 x i64>
-// CHECK:           %[[VAL_7:.*]] = llvm.mlir.constant(1 : i64) : !llvm.i64
-// CHECK:           %[[VAL_8:.*]] = llvm.mlir.constant(1 : i16) : !llvm.i1
-// CHECK:           %[[VAL_9:.*]] = llvm.icmp "eq" %[[VAL_5]], %[[VAL_8]] : !llvm.i1
-// CHECK:           llvm.cond_br %[[VAL_9]], ^bb1, ^bb2
+// CHECK:           %[[VAL_5:.*]] = llvm.mlir.constant(0 : i32) : !llvm.i32
+// CHECK:           %[[VAL_6:.*]] = llvm.mlir.constant(1 : i32) : !llvm.i32
+// CHECK:           %[[VAL_7:.*]] = llvm.getelementptr %[[VAL_4]]{{\[}}%[[VAL_5]], %[[VAL_5]]] : (!llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>, !llvm.i32, !llvm.i32) -> !llvm.ptr<ptr<i8>>
+// CHECK:           %[[VAL_8:.*]] = llvm.load %[[VAL_7]] : !llvm.ptr<ptr<i8>>
+// CHECK:           %[[VAL_9:.*]] = llvm.getelementptr %[[VAL_4]]{{\[}}%[[VAL_5]], %[[VAL_6]]] : (!llvm.ptr<struct<(ptr<i8>, i64, i64, i64)>>, !llvm.i32, !llvm.i32) -> !llvm.ptr<i64>
+// CHECK:           %[[VAL_10:.*]] = llvm.load %[[VAL_9]] : !llvm.ptr<i64>
+// CHECK:           %[[VAL_11:.*]] = llvm.bitcast %[[VAL_8]] : !llvm.ptr<i8> to !llvm.ptr<i16>
+// CHECK:           %[[VAL_12:.*]] = llvm.load %[[VAL_11]] : !llvm.ptr<i16>
+// CHECK:           %[[VAL_13:.*]] = llvm.trunc %[[VAL_10]] : !llvm.i64 to !llvm.i16
+// CHECK:           %[[VAL_14:.*]] = llvm.lshr %[[VAL_12]], %[[VAL_13]] : !llvm.i16
```

Comment:
Random thought, but can the check lines in this testcase be trimmed down?  This looks like an exhaustive match of the output for a large test, it isn't a targeted test for specific functionality.

Reply:
That's a good point. Probably a lot of the tests in there can be simplified. For this drive condition test I reduced it to only check for the instructions that are inserted to manage this condition (done [here](https://github.com/llvm/circt/blob/ef14ef11aa6678f3affac51de6cbd2047de64742/lib/Conversion/LLHDToLLVM/LLHDToLLVM.cpp#L1618)), because for the default drive another test exists.